### PR TITLE
File system finalizer

### PR DESCRIPTION
**Pull request by arhimondr**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -305,24 +305,23 @@
...
-        <!-- discard all logging from hadoop -->
         <dependency>
             <groupId>org.slf4j</groupId>
-            <artifactId>jcl-over-slf4j</artifactId>
+            <artifactId>slf4j-api</artifactId>
             <version>${dep.slf4j.version}</version>
-            <scope>runtime</scope>
         </dependency>
 
+        <!-- discard all logging from hadoop -->
```
Comment (electrum) [inline pom.xml]: This comment seems incorrect now, since we're not discarding output. That's a big change (from a behavioral standpoint) and should be a separate commit. We'll need to tell users about this as they will likely want to change their log level configuration.

File: src/main/java/org/apache/hadoop/fs/FinalizerService.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.fs;
+
+import com.google.common.collect.Sets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
```
Comment (electrum) [inline src/main/java/org/apache/hadoop/fs/FinalizerService.java]: We should probably use the JCL (`org.apache.commons.logging`) version as that is what the rest of Hadoop uses (e.g., in `FileSystem`).

Comment (yarosa) [reply src/main/java/org/apache/hadoop/fs/FinalizerService.java]: Maybe it is not a nicest solution, but i decided to just use the logger from the `FileSystem`. It is `public`.

**Comment by BlueStalker** [comment]
Is there any plans on merging this and make a new rel of presto-hadoop-apache2

**Comment by electrum** [comment]
Merged, thanks!
